### ISSUE TITLE
Error compiling for Arduino Uno

### ISSUE THREAD

**Issue by VaderConsulting** [bremme/arduino-tm1637#50]
// ==============================INCLUDES=========================================
#include "SevenSegmentTM1637.h"
#include "SevenSegmentExtended.h"

// Date and time functions using a DS1307 RTC connected via I2C and Wire lib
#include <Wire.h>
#include "RTClib.h"
#include <time.h>

#include <MsTimer2.h>
#include <NSEncoder.h>

// ======================================================================
// Display
// Instantiation and pins configurations
// Pin 12 - > DIO
// Pin 13 - > CLK
SevenSegmentExtended display(13, 12);

C:\Users\windo\OneDrive\Documents\Arduino\libraries\SevenSegmentTM1637\src\SevenSegmentFun.cpp:187:80: error: no 'void SevenSegmentFun::bouncingBall(uint16_t, uint16_t, bool)' member function declared in class 'SevenSegmentFun'
 void  SevenSegmentFun::bouncingBall(uint16_t moves, uint16_t d, bool runForever) {
                                                                                ^
C:\Users\windo\OneDrive\Documents\Arduino\libraries\SevenSegmentTM1637\src\SevenSegmentFun.cpp: In member function 'void SevenSegmentFun::print4Bit(uint8_t, uint8_t, uint8_t)':
C:\Users\windo\OneDrive\Documents\Arduino\libraries\SevenSegmentTM1637\src\SevenSegmentFun.cpp:250:65: warning: unused parameter 'y' [-Wunused-parameter]
 void SevenSegmentFun::print4Bit( const uint8_t x, const uint8_t y, uint8_t symbol) {
                                                                 ^
exit status 1
Error compiling for board Arduino Uno.

**Comment by danielbernalb** [comment]
I have the same problem with Arduino UNO:
...........src\SevenSegmentFun.cpp:187:80: error: no 'void SevenSegmentFun::bouncingBall(uint16_t, uint16_t, bool)' member function declared in class 'SevenSegmentFun'
 void  SevenSegmentFun::bouncingBall(uint16_t moves, uint16_t d, bool runForever) {
                                                                                ^

**Comment by danielbernalb** [comment]
My temporal solution:
Erase SevenSegmentFun.cpp and SevenSegmentFun.h

**Comment by oum100** [comment]
I got the same error. This error was wrong typing of "bouncingBall" in file SevenSegmentFun.cpp in line 187.  It not the same as declare in SevenSegmentFun.h.   To fixed it just correct it to "bouchingBall".

**Comment by schumar** [comment]
This has already been fixed (in 6f4f084), but you have to manually download the current release ([v1.1.1](https://github.com/bremme/arduino-tm1637/releases/tag/v1.1.1))

**Comment by B0rax** [comment]
@bremme can we get a release here? currently the library from the Arduino library manager is broken as it can only install V1.1.0.

**Comment by bremme** [comment]
I haven't looked at this for a long time. I just updated the version number in the `library.properties` file to 1.1.1 as well. I guess that should fix the issue right @B0rax ?

**Comment by B0rax** [comment]
@bremme i think so, yes. There is also a PR with this fix (#57 )

**Comment by bremme** [comment]
Yeah I say that PR as well, but that also included some functional changes. So just made a new commit with the version bump. Gonna close this for now. Let me know if the issue still persists. And thx by the way!

**Comment by B0rax** [comment]
@bremme It doesn't look like it was fixed. The Arduino IDE still offer 1.1.0 as the latest release.
I sadly can not add any more expertise here. Maybe it needs to have a release and the update to the `library.properties` at the same time? I don't know.

There is a checklist from Arduino but I guess you know it better than me:
https://github.com/arduino/library-registry/blob/main/FAQ.md#how-can-i-publish-a-new-release-once-my-library-is-in-the-list

**Comment by bremme** [comment]
Shit, that's unfortunate. I haven't worked with Arduino for years, so need to check the documentation as well. Let see if I can find some time to fix these issues soon. For the time being you can always just download the library from Github and use it that way ofc.

**Comment by per1234** [comment]
>Maybe it needs to have a release and the update to the library.properties at the same time?

Not at the same time, but rather in the correct order. The value of the `version` property in `library.properties` must be bumped _prior_ to making the release. The Arduino Library Manager system will reject any release that has the same version value as a release that has already been indexed.

The 1.1.1 release was rejected because at the time of the release, the `version` value was still `1.1.0`:

https://github.com/bremme/arduino-tm1637/blob/v1.1.1/library.properties#L2

```text
version=1.1.0
```

The value was only bumped after the time of the release:

https://github.com/bremme/arduino-tm1637/commit/6357c4e4c0de116e388085f42858a08e1312ef61

The solution will be to make a 1.1.2 release, following the correct procedure:

1. Bump the value of the `version` property in the library's `library.properties` metadata file to `1.1.2`.
1. Create a new [release](https://docs.github.com/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository) or [tag](https://git-scm.com/docs/git-tag) in the library repository.
